core/trino-main/src/main/java/io/trino/execution/RenameTableTask.java

```diff
@@ -67,23 +68,17 @@ public String getName()
         QualifiedObjectName tableName = createQualifiedObjectName(session, statement, statement.getSource());
 
         if (metadata.isMaterializedView(session, tableName)) {
-            if (!statement.isExists()) {
-                throw semanticException(
-                        TABLE_NOT_FOUND,
-                        statement,
-                        "Table '%s' does not exist, but a materialized view with that name exists. Did you mean ALTER MATERIALIZED VIEW %s RENAME ...?", tableName, tableName);
-            }
-            return immediateVoidFuture();
+            throw semanticException(
+                    GENERIC_USER_ERROR,
```

Comment:
We probably should stick with `TABLE_NOT_FOUND`

Reply:
@sopel39  we had a discussion on this here https://github.com/trinodb/trino/pull/11555#discussion_r836132958